### PR TITLE
lightning: gcs reader seekend should has an positive offset

### LINKED ISSUE #41899: lightning: gcs reader seekend should has an positive offset
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. create br/gcs storage backend
2. use seekend
### 2. What did you expect to see? (Required)
get the positive offset
### 3. What did you see instead (Required)
get the negative offset
### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: close #41899 

Problem Summary:
gcs reader seekend should has an positive offset
### What is changed and how it works?
get the total size of the object then file open.
return the `totalSize+offset` instead of `offset`.
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/storage/gcs.go [hunk 1]
```diff
@@ -395,8 +397,7 @@ func (r *gcsObjectReader) Seek(offset int64, whence int) (int64, error) {
 		if offset >= 0 {
```
Comment (lance6716) [inline br/pkg/storage/gcs.go:397]: please add a test about Seek(0, End)

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: br/pkg/storage/gcs.go [hunk 2]
```diff
@@ -395,8 +397,7 @@ func (r *gcsObjectReader) Seek(offset int64, whence int) (int64, error) {
...
 			return 0, errors.Annotatef(berrors.ErrInvalidArgument, "Seek: offset '%v' should be negative.", offset)
 		}
-		// GCS supports `NewRangeReader(ctx, -10, -1)`, which means read the last 10 bytes.
-		realOffset = offset
+		realOffset = offset + r.totalSize
```
Comment (gozssky) [inline br/pkg/storage/gcs.go:400]: What will happen if `realOffset < 0`?

File: br/pkg/storage/gcs.go (hunk 1 above)
Comment (Leavrth) [reply br/pkg/storage/gcs.go:397]: done

File: br/pkg/storage/gcs.go (hunk 2 above)
Comment (Leavrth) [reply br/pkg/storage/gcs.go:400]: done

File: br/pkg/storage/gcs_test.go [hunk 3]
```diff
@@ -226,6 +226,11 @@ func TestGCS(t *testing.T) {
 	require.Equal(t, 5, n)
 	require.Equal(t, "97222", string(p))
 
+	_, err = efr.Seek(int64(0), io.SeekEnd)
+	require.Error(t, err)
```
Comment (lance6716) [inline br/pkg/storage/gcs_test.go:230]: This should be the file size

**Review by gozssky — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Pull request merged**